core/chain_makers.go

```diff
@@ -131,6 +131,14 @@ func (b *BlockGen) PrevBlock(index int) *types.Block {
 	return b.chain[index]
 }
 
+// OffsetTime modifies the time instance of a block, implicitly changing its
+// associated difficulty. It's useful to test scenarios where forking is not
+// tied to chain length directly.
+func (b *BlockGen) OffsetTime(seconds int64) {
+	b.header.Time.Add(b.header.Time, new(big.Int).SetInt64(seconds))
```

Comment:
BlockGen also panics for invalid transactions. The idea is that you really don't want to debug failed insertions when writing a test for something else.